ewatercycle/models/pcrglobwb.py

```diff
@@ -15,94 +17,175 @@
+            e.g. `meteoOptions = {"temperatureNC": "era5_tas_1990_2000.nc"}`
+            where meteoOptions is the section in which the temperatureNC option
+            may be found. See :py:attr:`~parameters` for all available settings.
+
+        Returns: Path to config file and work dir
+        """
+        self._update_config(**kwargs)
+        cfg_file = self._export_config()
+        work_dir = self.work_dir
+
+        self._start_container()
+
+        return cfg_file, work_dir
+
+    def _update_config(self, **kwargs):
+        cfg = self.config
+
+        default_input_dir = self.parameter_set.input_dir
+
         for section, options in kwargs.items():
+            if not cfg.has_section(section):
+                cfg.add_section(section)
+
             for option, value in options.items():
-                cfg.set(section, option, value)
 
+                if Path(value).exists():
+                    # New data paths must be mounted on the container
+                    inputpath = Path(value).expanduser().resolve()
+                    if default_input_dir in inputpath.parents:
+                        pass
+                    elif inputpath.is_dir():
+                        self._additional_input_dirs.append(str(inputpath))
+                    else:
+                        self._additional_input_dirs.append(
+                            str(inputpath.parent))
+                    cfg.set(section, option, str(inputpath))
+                else:
+                    cfg.set(section, option, value)
+
+    def _export_config(self) -> PathLike:
         new_cfg_file = Path(self.work_dir) / "pcrglobwb_ewatercycle.ini"
         with new_cfg_file.open("w") as filename:
-            cfg.write(filename)
+            self.config.write(filename)
 
-        self.cfg_file = new_cfg_file.resolve()
-        print(f"Created config file {self.cfg_file} with inputDir "
-              f"{full_input_path} and outputDir {self.work_dir}.")
+        self.cfg_file = new_cfg_file.expanduser().resolve()
+        return self.cfg_file
 
-    def _start_container(self,
-                         input_dir: PathLike,
-                         additional_input_dirs: Iterable[PathLike] = []):
-        input_dirs = [input_dir] + list(additional_input_dirs)
+    def _start_container(self):
+        input_dirs = [self.parameter_set.input_dir
+                      ] + self._additional_input_dirs
 
         if CFG["container_engine"] == "docker":
             self.bmi = BmiClientDocker(
-                image=CFG["pcrglobwb.docker_image"],
+                image=self.docker_image,
                 image_port=55555,
                 work_dir=str(self.work_dir),
                 input_dirs=[str(input_dir) for input_dir in input_dirs],
                 timeout=10,
             )
         elif CFG["container_engine"] == "singularity":
-            image = CFG["pcrglobwb.singularity_image"]
-
             message = f"No singularity image found at {image}"
```

Comment:
image var is not available 
```suggestion
            message = f"No singularity image found at {self.singularity_image}"
```